driver/conv_driver.hpp

```diff
@@ -201,16 +210,197 @@ bool readBufferFromFile(T* data, size_t dataNumItems, const char* fileName)
     }
 }
 
+template <typename Tgpu>
+class GpumemTensor
+{
+    std::unique_ptr<GPUMem> dev;
+    tensor<Tgpu> host;
+    bool is_gpualloc = false;
+
+public:
+    void SetGpuallocMode(bool v) { is_gpualloc = v; }
+    tensor<Tgpu>& GetTensor() { return host; }
+
+    void AllocOnHost(miopenTensorDescriptor_t t)
+    {
+        host = tensor<Tgpu>(miopen::deref(t));
+        if(is_gpualloc) // We do not need host data.
+        {
+            host.data.clear();
+            host.data.shrink_to_fit(); // To free host memory.
+        }
+    }
```

Comment:
What shall we do in an `else` case?
Its name is `AllocOnHost` but it does deallocation for `is_gpualloc` or does nothing. That may be confusing.

Reply:
In gpualloc mode it does nothing because we do not need buffers on the host. And yes, the design may confuse the reader, but the root reason of this is clumsy initial design of the driver. At least we have a kind of imperfect abstraction now; let's improve this stuff in the future.

Reply:
[Gentle reminder] @CAHEK7 Please mark this review thread as "resolved" or provide some other kind of feedback.

Reply:
@CAHEK7 Yes. The goal of this PR is only introducing the new functionality. Before this change we had one big bucket of worms, now we have two smaller ones. Some progress as a by-product, right?

Reply:
Kind of, but why should we create two buckets of worms and a cup instead of getting rid of cup right now. Why we should introduce a new functionality which is kind of dangerous from the beginning?

Reply:
@CAHEK7 The danger with metaphors is that they can be easily misunderstood. This PR does not _create_ a bucket of worms. It sorts existing worms into buckets, making it easier to get rid of them in the future. The fewer worms in a bucket, the easier it is to notice them, and the easier it is to get rid of them (the easiest way is if each worm sits in its own bucket).

But maybe this metaphor is completely inappropriate for this case. AFAICS the driver needs to be redesigned, but that work is clearly out of the scope or this PR and does not fit into my mind anyway 🤷‍♂️ 

Thanks for reviewing.